File: Data Structures and Algorithms/Prefinals/StudentList/student_list.py
def main() -> None:
    CLASSMATE_COUNT = 3
    students = {}
    third_entry_key = None
    for i in range(1, CLASSMATE_COUNT + 1):
        key = input(f"Enter student number {i}: ")
        students[key] = input(f"Enter first name {i}: ")
        if i == 3:
            third_entry_key = key
    print_student_list(students)

    del students[third_entry_key]
    students[input("Enter your student number: ")] = input("Enter your first name")
    print_student_list(students)


def print_student_list(students: dict) -> None:
    print("Student List:")
    for key in students:
        print(f"{key} {students[key]}")


if __name__ == "__main__":
    main()

File: Data Structures and Algorithms/Prefinals/StudentList/StudentList.java
import java.util.Scanner;
import java.util.Map;
import java.util.HashMap;

public class StudentList {
    private static final Scanner sc = new Scanner(System.in);
    private static Map<String, String> students = new HashMap<>();

    public static void main(String[] args) {
        String thirdEntryKey = null;
        for (int i = 1, classmates = 3; i <= classmates; i++) {
            String key = input(String.format("Enter student number %s: ", i));
            students.put(key, input(String.format("Enter first name %s: ", i)));
            if (i == 3) {
                thirdEntryKey = key;
            }
        }

        printStudentList(students);
        students.remove(thirdEntryKey);
        students.put(input("Enter your student number: "), input("Enter your first name: "));
        printStudentList(students);
    }

    private static void printStudentList(Map<String, String> students) {
        System.out.println("Student List:");
        for (String key : students.keySet()) {
            System.out.printf("%s %s\n", key, students.get(key));
        }
    }

    private static String input(String prompt) {
        System.out.print(prompt);
        return sc.nextLine();
    }
}